apps/api/src/app/shared/services/query-parser/query-validator.service.ts

```diff
@@ -0,0 +1,245 @@
+        if (!value) {
+          issues.push({
+            message: 'Field variable is required',
+            path,
+            type: QueryIssueTypeEnum.MISSING_VALUE,
+          });
+        }
+
+        continue;
+      }
+    }
+  }
+
+  private validateBetweenOperation({
+    value,
+    issues,
+    path,
+  }: {
+    value: [unknown, unknown, unknown];
+    issues: QueryIssue[];
+    path: number[];
+  }) {
+    const [lowerBound, field, upperBound] = value;
+
+    if (this.isInvalidFieldReference(field)) {
+      issues.push(this.getFieldReferenceIssue(path));
+    }
+
+    if (lowerBound === undefined || lowerBound === null || upperBound === undefined || upperBound === null) {
+      issues.push(this.getValueIssue(path));
+    }
+  }
+
+  private validateComparisonOperation({
+    operator,
+    value,
+    issues,
+    path,
+  }: {
+    operator: JsonComparisonOperatorEnum;
+    value: unknown;
+    issues: QueryIssue[];
+    path: number[];
+  }) {
+    if (!Array.isArray(value) || value.length !== 2) {
+      issues.push(this.getOperationIssue(operator, path));
+
+      return;
+    }
+
+    const [field, comparisonValue] = value;
+
+    // Validate field reference
+    if (this.isInvalidFieldReference(field)) {
+      issues.push(this.getFieldReferenceIssue(path));
+    }
+
+    // Validate comparison value exists
+    const valueIsUndefinedOrEmptyCase =
+      (comparisonValue === undefined || comparisonValue === '') &&
+      [
+        JsonComparisonOperatorEnum.EQUAL,
+        JsonComparisonOperatorEnum.NOT_EQUAL,
+        JsonComparisonOperatorEnum.LESS_THAN,
+        JsonComparisonOperatorEnum.GREATER_THAN,
+        JsonComparisonOperatorEnum.LESS_THAN_OR_EQUAL,
+        JsonComparisonOperatorEnum.GREATER_THAN_OR_EQUAL,
+        JsonComparisonOperatorEnum.STARTS_WITH,
+        JsonComparisonOperatorEnum.ENDS_WITH,
+      ].includes(operator);
+    const valueIsNullCase =
+      comparisonValue === null &&
+      ![JsonComparisonOperatorEnum.EQUAL, JsonComparisonOperatorEnum.NOT_EQUAL].includes(operator);
+
+    if (valueIsUndefinedOrEmptyCase || valueIsNullCase) {
+      issues.push(this.getValueIssue(path));
+    }
+
+    // Validate array for 'in' operations
+    if (operator === 'in' && !Array.isArray(comparisonValue)) {
+      issues.push(this.getOperationIssue(operator, path));
+    }
+  }
+
+  /*
+   * in operator has field and the array as operands
+   * but as contains it has the search value and the field as operands
+   */
+  private validateInOperation({ value, issues, path }: { value: unknown; issues: QueryIssue[]; path: number[] }) {
+    if (!Array.isArray(value) || value.length !== 2) {
+      issues.push(this.getOperationIssue('in', path));
+
+      return;
+    }
+
+    const [firstOperand, secondOperand] = value;
+    const isContains = typeof firstOperand === 'string';
+
+    if (isContains) {
+      // Validate search value exists
+      if (firstOperand === undefined || firstOperand === '') {
```

Comment:
Just a personal preference, but can also be:
```suggestion
      const searchValueExists = firstOperand === undefined || firstOperand === '';
      if (searchValueExists) {
```

Reply:
Same for some of the rest of the comments